include/boost/math/distributions/dirichlet.hpp

```diff
@@ -248,9 +224,27 @@ class dirichlet_distribution
     }
   } // void find_alpha
 
+  RealType normalizing_constant(RealType b = 0.0) const
+  {
+    // B(a1,a2,...ak) = tgamma(a1+a2+...+ak)/(tgamma(a1)*tgamma(a2)...*tgamma(ak)
+    RealType mb;
+    RealType alpha_sum = accumulate(m_alpha.begin(), m_alpha.end(), b * m_alpha.size());
+    for (decltype(m_alpha.size()) i = 0; i < m_alpha.size(); ++i)
+    {
+      mb *= tgamma(m_alpha[i] + b);
+    }
+    mb /= tgamma(alpha_sum);
+    return mb;
+  } // normalizing_constant
+
+  RealType sum_alpha() const
+  {
+    return accumulate(m_alpha.begin(), m_alpha.end(), 0);
```

Comment:
Should this be `std::accumulate`? `std::accumulate` works on any type with `+` defined; it doesn't need to use ADL for multiprecision types.